src/stores.js

```diff
@@ -4,248 +4,81 @@
 //
 // This file contains the model state for all the cards.
 
+const EosDiscoveryFeed = imports.gi.EosDiscoveryFeed;
 const Gio = imports.gi.Gio;
 const GObject = imports.gi.GObject;
 
 const Lang = imports.lang; 
 
-var CARD_STORE_TYPE_ARTICLE_CARD = 0;
-var CARD_STORE_TYPE_WORD_QUOTE_CARD = 1;
-var CARD_STORE_TYPE_ARTWORK_CARD = 2;
-var CARD_STORE_TYPE_AVAILABLE_APPS = 3;
-var CARD_STORE_TYPE_VIDEO_CARD = 4;
-var CARD_STORE_TYPE_NEWS_CARD = 5;
+var CARD_STORE_TYPE_ARTICLE_CARD = EosDiscoveryFeed.CardStoreType.ARTICLE_CARD;
+var CARD_STORE_TYPE_WORD_QUOTE_CARD = EosDiscoveryFeed.CardStoreType.WORD_QUOTE_CARD;
+var CARD_STORE_TYPE_ARTWORK_CARD = EosDiscoveryFeed.CardStoreType.ARTWORK_CARD;
+var CARD_STORE_TYPE_AVAILABLE_APPS = EosDiscoveryFeed.CardStoreType.AVAILABLE_APPS;
+var CARD_STORE_TYPE_VIDEO_CARD = EosDiscoveryFeed.CardStoreType.VIDEO_CARD;
+var CARD_STORE_TYPE_NEWS_CARD = EosDiscoveryFeed.CardStoreType.NEWS_CARD;
 var CARD_STORE_TYPE_MAX = CARD_STORE_TYPE_NEWS_CARD;
 
 // eslint-disable-next-line no-unused-vars
 var DiscoveryFeedCardStore = new Lang.Class({
     Name: 'DiscoveryFeedCardStore',
-    Extends: GObject.Object,
-    Properties: {
-        type: GObject.ParamSpec.int('type',
-                                    '',
-                                    '',
-                                    GObject.ParamFlags.READWRITE |
-                                    GObject.ParamFlags.CONSTRUCT_ONLY,
-                                    CARD_STORE_TYPE_ARTICLE_CARD,
-                                    CARD_STORE_TYPE_MAX,
-                                    CARD_STORE_TYPE_ARTICLE_CARD)
-    }
+    Extends: EosDiscoveryFeed.BaseCardStore
 });
```

Comment:
Obviously, using `Extends` here is just transitional - my plan is to either remove/alias these alltogether.

Reply:
What I meant is to try and remove this transitional churn before merging this PR.

Reply:
Ah got it. I actually removed it in a subsequent PR to this one, but I can backport the relevant part of the commit

Reply:
Okay, I've cleaned up the transitional churn to the extent that I removed all the store definitions. I don't really have time to deal with cleaning up all the constants today, but a lot of the usage of those constants got removed in the subsequent PR anyway, so I'd prefer to leave them alone to avoid double-work